src/pelicanfs/token_content_iterator.py

```diff
@@ -82,21 +96,203 @@ class TokenContentIterator:
+                                break
+                            try:
+                                if not read_and_echo_output(master_fd):
+                                    break
+                            except OSError:
+                                break
+                        break
+
+                    # Wait for data from either master_fd (subprocess) or stdin (user input)
+                    # In Jupyter/IPython, sys.stdin may not support fileno(), so we only monitor master_fd
+                    if stdin_is_real:
+                        # Real stdin - monitor it with select
+                        fds_to_monitor = [master_fd, sys.stdin]
+                    else:
+                        # Redirected stdin - only monitor master_fd
+                        fds_to_monitor = [master_fd]
+
+                    ready_read, _, _ = select.select(fds_to_monitor, [], [], SELECT_TIMEOUT)
+
+                    for fd in ready_read:
+                        try:
+                            if fd == master_fd:
+                                if not read_and_echo_output(master_fd):
+                                    break
+                            elif stdin_is_real and fd == sys.stdin:
+                                # Data from real stdin - forward to subprocess
+                                data = os.read(stdin_fd, PTY_BUFFER_SIZE)
+                                if data:
+                                    os.write(master_fd, data)
+                        except OSError:
+                            break
+
+                    # If stdin is redirected and we have data to send, write it to the subprocess
+                    if not stdin_is_real and stdin_data_to_send:
+                        try:
+                            os.write(master_fd, stdin_data_to_send.encode("utf-8"))
+                            stdin_data_to_send = None  # Only send once
+                        except OSError as e:
+                            logger.debug(f"Error writing redirected stdin to PTY: {e}")
+            finally:
+                os.close(master_fd)
+
+            # Ensure process has finished
+            process.wait()
+
+            if process.returncode != 0:
+                logger.debug(f"Pelican binary exited with code {process.returncode}")
+                return None
+
+            # Extract JWT token from captured output
+            full_output = b"".join(output_data).decode("utf-8", errors="replace")
+            jwt_pattern = r"eyJ[A-Za-z0-9_-]+\.[A-Za-z0-9_-]+\.[A-Za-z0-9_-]+"
+            matches = re.findall(jwt_pattern, full_output)
+
+            if matches:
+                token = matches[-1]
+                logger.info("Successfully acquired token via OIDC device flow")
+                return token
+            else:
+                logger.warning("Could not extract JWT token from pelican binary output")
+                logger.debug(f"Output was: {full_output[:500]}")  # Log first 500 chars
```

Comment:
Is there anything that can be done to hook the logging levels from PelicanFS into the binary? For example, let's say I try to use this OIDC flow to handle tokens, but the golang binary chokes on something and I want to get its debug output to see if there's an obvious issue. Do I as a user of PelicanFS have to figure out how to pass debug levels via some side channel to the go binary? Can you extend the PelicanFS log level to determine whether the `-d` flag should also be passed using your `_get_pelican_flag()` method?

Also, why only log the first 500 chars? The comment doesn't explain why that decision is being made.